internal/report/json/json.go

```diff
@@ -25,51 +25,35 @@ func (r *Reporter) Add(t report.TestResult) {
 
 // Render sends the result to specified webhook WebhookURL and log the result to the specified json file
 func (r *Reporter) Render() {
-	r.cleanup()
 	body, err := json.Marshal(r.Results)
 	if err != nil {
-		log.Error().Msgf("failed to generate test result (%v)", err)
+		log.Err(err).Msg("failed to generate test result.")
 		return
 	}
 
 	if r.WebhookURL != "" {
 		resp, err := http.Post(r.WebhookURL, "application/json", bytes.NewBuffer(body))
 		if err != nil {
-			log.Error().Err(err).Str("webhook", r.WebhookURL).Msg("failed to send test result to webhook.")
+			log.Err(err).Str("webhook", r.WebhookURL).Msg("failed to send test result to webhook.")
 		} else {
 			webhookBody, _ := io.ReadAll(resp.Body)
 			if resp.StatusCode >= http.StatusBadRequest {
-				log.Error().Str("webhook", r.WebhookURL).Msgf("failed to send test result to webhook, status: '%d', msg:'%v'", resp.StatusCode, string(webhookBody))
+				log.Error().Str("webhook", r.WebhookURL).Msgf("failed to send test result to webhook, status: %d, msg: %q.", resp.StatusCode, string(webhookBody))
 			}
 			if resp.StatusCode%100 == 2 {
-				log.Info().Str("webhook", r.WebhookURL).Msgf("test result has been sent successfully to webhook, msg: '%v'.", string(webhookBody))
+				log.Info().Str("webhook", r.WebhookURL).Msgf("test result has been sent successfully to webhook, msg: %q.", string(webhookBody))
 			}
 		}
 	}
 
 	if r.Filename != "" {
 		err = os.WriteFile(r.Filename, body, 0666)
 		if err != nil {
-			log.Error().Err(err).Msgf("failed to write test result to %s", r.Filename)
+			log.Err(err).Msgf("failed to write test result to %s.", r.Filename)
 		}
 	}
 }
 
-// cleanup removes any information that isn't relevant in the rendered report. Particularly when it comes to
-// artifacts, this reporter is only interested in those that have been persisted to the file system.
-func (r *Reporter) cleanup() {
```

Comment:
Is this behavior not needed anymore?

Reply:
It's not needed anymore do to the changes in https://github.com/saucelabs/saucectl/pull/849/files#diff-ea42af98511d2bfb326add8027a845ef0e7cfe15c81221258fdf22fd7367df6fR367

RDC doesn't have this issue because it appends artifacts after the error handling, which means there is no empty `FilePath`. https://github.com/saucelabs/saucectl/blob/main/internal/http/rdcservice.go#L396